lib/hashie/extensions/parsers/yaml_erb_parser.rb

```diff
@@ -11,6 +11,7 @@ def initialize(file_path)
 
         def perform
           template = ERB.new(@content)
+          @file_path = @file_path.to_s if @file_path.respond_to? :to_s
```

Comment:
I don't think this is the right place to do this, if `perform` is called over and over again it's wasteful, and it's modifying the state of the object which feels like a side effect. Furthermore if I pass some junk that responds to `to_s` here we'll just try to load that, and that doesn't seem safe. 

Ideally this should be directly supported by `YAML.load`, but short of that I would do this in `initialize` and more explicitly.

```ruby
@file_path = file_path.is_a?(Pathname) ? file_path.to_s : file_path
```